packages/blueprints/blueprint/src/blueprint.ts

```diff
@@ -318,3 +315,74 @@ function structureExistingBlueprints(location: string | undefined): BlueprintIns
   }
   return [];
 }
+
+/**
+ * This generates the default PR information that resynth updates will go onto.
+ */
+function defaultResynthPR(params: { options: any; context: Context }): {
+  branch: string;
+  title: string;
+  description: string;
+} {
+  if (params.context.project.blueprint.instantiationId) {
+    // The blueprint options/versions are being changed.
+    return {
+      branch: `update-blueprint-${formatBlueprintName(
+        params.context.package.name || 'unknown',
+      )}-${params.context.project.blueprint.instantiationId.slice(0, 10)}`,
+      title: `chore(resynthesis): update [${params.context.package.name}@${params.context.package.version}]`,
+      description: [
+        'This is a pull request created from the options/versions being changed on your blueprint instance.',
+        '',
+        `Blueprint: [${params.context.package.name}@${params.context.package.version}]`,
+        `Instance Id: [${params.context.project.blueprint.instantiationId}]`,
+        '### New Options',
+        '```',
+        JSON.stringify(params.options, null, 2),
+        '```',
+      ].join('\n'),
+    };
+  } else {
+    // The blueprint has no existing instantiationId and is being applied for the first time.
+    return {
+      branch: `apply-blueprint-${formatBlueprintName(params.context.package.name || 'unknown')}-${hash(Date.now().toString())}`,
+      title: `feat: application [${params.context.package.name}@${params.context.package.version}]`,
+      description: [
+        'This is the initial pull request created from a new blueprint instance being applied to your project.',
+        '',
+        `Blueprint: [${params.context.package.name}@${params.context.package.version}]`,
+        '### Selected Options',
+        '```',
+        JSON.stringify(params.options, null, 2),
+        '```',
+      ].join('\n'),
+    };
+  }
+}
+
+/**
+ * Does custom formatting of the blueprint name to make it more human readable.
+ * By default, blueprint package names come in the form: "@amazon-codecatalyst/blueprints.sam-serverless-application"
+ * The default form is too long, this function formats that to something like sam-serverless-application
+ * @param packageName - package name
+ * @returns string
+ */
+function formatBlueprintName(packageName: string, replacement: string = '-'): string {
+  const parseFromLast = (input: string, char: string): string => {
+    const lastIndex = input.lastIndexOf(char);
+    if (lastIndex === -1) {
+      return input;
+    }
+    return input.slice(lastIndex + 1);
+  };
+
+  return parseFromLast(packageName, '.').replace(/[^a-zA-Z]/g, replacement);
```

Comment:
nit: npm packages with numbers are valid, so this would convert a package like `my-blueprint-123` to `my-blueprint----`. Valid as a branch name, but still a bit weird.

Reply:
Thats a good point, I shouldnt nuke numbers. I'll update